napari_crop/_function.py

```diff
@@ -74,9 +76,9 @@ def crop_region(
             # broadcast the mask to the shape of the cropped image
             mask = np.broadcast_to(mask_nD, cropped_data.shape)
             cropped_data[~mask] = 0
-
-    layer_props["name"] = layer_props["name"] + " (cropped)"
-    if layer_type == "image":
-        return napari.layers.Image(cropped_data, **layer_props)
-    if layer_type == "labels":
-        return napari.layers.Labels(cropped_data, **layer_props)
+        layer_props["name"] = layer_props["name"] + " (cropped)"
+        if layer_type == "image":
+            cropped_list.append((cropped_data, layer_props, 'image'))
+        if layer_type == "labels":
+            cropped_list.append((cropped_data, layer_props, 'labels'))
```

Comment:
```suggestion
        new_layer_props = layer_props.copy()
        new_layer_props["name"] = layer_props["name"] + f" (cropped {count+1})"
        cropped_list.append((cropped_data, new_layer_props, layer_type))
```
It turns out, we cannot pass the same identical dictionary to all new layers. It must be copies. The underlying reason is: if there are two layers with the same name, the data will be stored to the same layer. That's why I was adding a number to the name.
Furthermore, the two `if`-statements were not necessary. We can pass the type-string as parameter as it is.

Reply:
Nice @haesleinhuepf ! It works here!
I am going to commit changes and, if that's OK, merge it afterwards.